pkg/component/component.go

```diff
@@ -68,6 +68,31 @@ func CreateFromGit(client *occlient.Client, name string, ctype string, url strin
 	return nil
 }
 
+// GetComponentPorts provides slice of ports used by the component in the form port_no/protocol
+func GetComponentPorts(client *occlient.Client, componentName string, applicationName string) (res []string, err error) {
```

Comment:
Hate to say this, but you'll need to add a test for this function :smile: 

If you need help, feel free to PM / ping me or @ashetty1  / @mik-dass  ! I just finished doing #565 and adding tests there and had to learn a bunch of new stuff.

Reply:
Ack